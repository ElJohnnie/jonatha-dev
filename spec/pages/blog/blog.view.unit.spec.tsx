import React from 'react';
import { render, screen } from '@testing-library/react';
import BlogView from '@/app/[locale]/blog/view/blog.view';

describe('BlogView', () => {
  it('deve renderizar o título do blog', () => {
    const { container } = render(<BlogView>Conteúdo de teste</BlogView>);

    const titleElement = screen.getByText('Blog');

    expect(titleElement).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('deve renderizar o conteúdo filho', () => {
    render(
      <BlogView>
        <div data-testid='child'>Filho</div>
      </BlogView>
    );

    const childElement = screen.getByTestId('child');

    expect(childElement).toBeInTheDocument();
  });
});
