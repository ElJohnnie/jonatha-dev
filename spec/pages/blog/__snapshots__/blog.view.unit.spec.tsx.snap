// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogView deve renderizar o título do blog 1`] = `
<section
  class="h-100 mt-2 pt-2"
>
  <div
    class="mx-auto max-w-7xl justify-center px-4 md:py-12"
  >
    <div
      class="mx-auto lg:mx-0"
    >
      <h1
        class="mb-4 max-w-2xl text-3xl font-extrabold leading-none tracking-tight md:text-6xl dark:text-white"
      >
        <span
          class="leading-12 block w-full bg-clip-text py-2 text-slate-600 lg:inline"
        >
          Blog
        </span>
      </h1>
    </div>
    <div
      class="mx-auto mt-6 grid max-w-2xl grid-cols-1 gap-x-8 gap-y-16 border-t border-gray-200 pt-6 sm:mt-10 sm:pt-10 lg:mx-0 lg:max-w-none lg:grid-cols-3"
    >
      Conteúdo de teste
    </div>
  </div>
</section>
`;
